src/main/java/eu/xenit/gradle/docker/compose/DockerComposeConventionImpl.java

```diff
@@ -108,23 +129,30 @@ public void fromProject(Project project) {
                 composeTask.doFirst(CONFIGURE_COMPOSE_ACTION_NAME, new Action<Task>() { // No lambda -> see #96
                     @Override
                     public void execute(Task t) {
-                        createSetComposeEnvironmentFromPathAction().execute(dockerBuildImage);
+                        if(environmentVariable != null) {
+                            String addition = "_" + dockerBuildImage.getName();
+                            String safeEnvironmentVariable = Util.safeEnvironmentVariableName(environmentVariable + addition);
```

Comment:
```suggestion
                            String safeEnvironmentVariable = environmentVariable + Util.safeEnvironmentVariableName(addition);
```

`saveEnvironmentVariableName()` should not be applied to the user-specified portion of the environment variable.